geoips/image_utils/mpl_utils.py

```diff
@@ -772,3 +774,123 @@ def create_colorbar(fig, mpl_colors_info):
         cbar.set_label(cbar_label, **set_label_kwargs)
 
     return cbar
+
+
+def add_shape_patches(main_ax, data_dict, mapobj, shape_type, cmap):
+    """
+    Add patches via data_dict in the form of shape_type to the matplotlib "main_ax".
+
+    Parameters
+    ----------
+    main_ax : Axes
+        matplotlib Axes object for plotting data and overlays
+    data : dict
+        Dict of numpy arrays to plot. Keys:
+        {"latitude": data, "longitude": data, "product_name": data}
+    mapobj : Map Object
+        Basemap or Cartopy CRS instance
+    shape_type : str
+        Specifies matplotlib Colors parameters for use in both plotting and colorbar
+    cmap: Matplotlib Colormap
+        Specificies the colormap to be used when coloring patches
+
+    See Also
+    --------
+    geoips.image_utils.mpl_utils.create_colorbar
+        for field descriptions for mpl_colors_info
+    """
+    import cartopy.crs as ccrs
+    import inspect
+
+    available_patches = []
+
+    for patch_class in inspect.getmembers(matplotlib.patches, inspect.isclass):
+        if "matplotlib.patches." in str(patch_class[1]) and "Style" not in str(
+            patch_class[0]
+        ):
+            available_patches.append(patch_class[0])
+
+    shape_type = shape_type.title().replace("_", "")
+    if shape_type not in available_patches:
+        raise TypeError(
+            f"Error {shape_type} isn't in the list of valid patches {available_patches}"
+        )
+
+    LOG.info(f"{shape_type} Matplotlib Patches being applied.")
+
+    patch = getattr(matplotlib.patches, shape_type)
+    crs = ccrs.CRS(mapobj)
+    norm = matplotlib.colors.Normalize(vmin=0, vmax=25)
+    LOG.info(f"Latitude size = {data_dict['latitude'].size}")
+    LOG.info(f"Longitude size = {data_dict['longitude'].size}")
+    LOG.info(f"Product size = {data_dict['product'].size}")
+    for idx in range(data_dict["latitude"].size):
+        lat = data_dict["latitude"][idx]
+        lon = data_dict["longitude"][idx]
+        product_val = data_dict["product"][idx]
+        shape = add_patch(
+            lat,
+            lon,
+            product_val,
+            patch,
+            crs,
+            cmap,
+            norm,
+            mapobj,
+        )
+        main_ax.add_patch(shape)
+    return main_ax
+
+
+def add_patch(lat, lon, product_val, patch, crs, cmap, norm, mapobj):
```

Comment:
We should talk about this function more to see if we want to add additional functionality for different shapes of patches. Could be useful for wind barbs, atmospheric rivers, etc.

Reply:
Issue #410 identifies the need for discussion on this point.

Reply:
Closing this comment as the issue above addresses it.